src/main/kotlin/com/sourcegraph/cody/history/HistoryTree.kt

```diff
@@ -86,11 +87,21 @@ class HistoryTree(
   }
 
   private fun deleteSelected() {
-    tree.selectedLeafOrNull()?.let { leaf ->
-      onDelete(leaf.chat)
-      val period = leaf.parent as PeriodNode
-      model.removeNodeFromParent(leaf)
+    tree.selectedLeafOrNull()?.let { selectedLeaf ->
+      val period = selectedLeaf.parent as PeriodNode
+      val selectedIndex = period.getIndex(selectedLeaf)
+      onDelete(selectedLeaf.chat)
+      model.removeNodeFromParent(selectedLeaf)
       if (model.getChildCount(period) == 0) model.removeNodeFromParent(period)
+
+      val leafsCount = period.childCount
```

Comment:
This logic will preserve selection:
* on the same index if possible
* on the previous index if removed element was the last one